spring-cloud-deployer-spi/src/main/java/org/springframework/cloud/deployer/spi/app/AbstractActuatorTemplate.java

```diff
@@ -66,55 +82,80 @@ protected AbstractActuatorTemplate(RestTemplate restTemplate, AppDeployer appDep
+
+		String actuatorUrl = getActuatorUrl(appInstanceStatus);
+
+		HttpHeaders requestHeaders = requestHeaders(httpHeadersForInstance(appInstanceStatus), optionalRequestHeaders);
 
 		ResponseEntity<R> responseEntity = httpPost(UriComponentsBuilder
-				.fromHttpUrl(actuatorUrl).path(normalizePath(endpoint)).toUriString(), body, responseType);
+				.fromHttpUrl(actuatorUrl).path(normalizePath(endpoint)).toUriString(), body, responseType,
+				requestHeaders);
 		if (responseEntity.getStatusCode().isError()) {
 			logger.error(responseEntity.getStatusCode().toString());
 		}
 		return responseEntity.getBody();
 	}
 
-	protected final HttpHeaders requestHeaders() {
-		HttpHeaders headers = new HttpHeaders();
-		headers.setAccept(Collections.singletonList(MediaType.APPLICATION_JSON));
-		headers.setContentType(MediaType.APPLICATION_JSON);
-		this.httpHeadersConsumer.accept(headers);
-		return headers;
+	protected final String getActuatorUrl(AppInstanceStatus appInstanceStatus) {
+		try {
+			return actuatorUrlForInstance(appInstanceStatus);
+		} catch (Exception e) {
+			throw new IllegalArgumentException(String.format(
+					"Unable to determine actuator url for app with guid %s",
+					appInstanceStatus.getAttributes().get("guid")));
+		}
 	}
 
-	protected final void setHttpHeadersConsumer(Consumer<HttpHeaders> httpHeadersConsumer) {
-		Assert.notNull(httpHeadersConsumer, "'httpHeadersConsumer cannot be null");
-		this.httpHeadersConsumer = httpHeadersConsumer;
+	protected abstract String actuatorUrlForInstance(AppInstanceStatus appInstanceStatus);
+
+	/**
+	 * Hook to allow subclasses to add special headers derived {@link AppInstanceStatus} metadata if necessary.
+	 *
+	 * @param appInstanceStatus the AppInstanceStatus for the target instance.
+	 * @return HttpHeaders
+	 */
+	protected Optional<HttpHeaders> httpHeadersForInstance(AppInstanceStatus appInstanceStatus) {
+		return Optional.empty();
 	}
 
-	private final String normalizePath(String path) {
-		return path.startsWith("/") ? path : "/" + path;
+
+	private final HttpHeaders requestHeaders(Optional<HttpHeaders> optionalAppInstanceHeaders,
+			Optional<HttpHeaders> optionalRequestHeaders) {
+		HttpHeaders requestHeaders = optionalAppInstanceHeaders
+				.orElse(new HttpHeaders());
+		optionalRequestHeaders.ifPresent(requestHeaders::addAll);
+
+		//Any pass-thru auth overrides the default.
+		if (!requestHeaders.containsKey(HttpHeaders.AUTHORIZATION)) {
+			addDefaultAuthorization(requestHeaders);
+		}
+
+		requestHeaders.setAccept(Collections.singletonList(MediaType.APPLICATION_JSON));
+		requestHeaders.setContentType(MediaType.APPLICATION_JSON);
+		return requestHeaders;
 	}
 
-	private final <T> ResponseEntity<T> httpGet(String url, Class<T> responseType) {
-		return restTemplate.exchange(url, HttpMethod.GET,
-				new HttpEntity(requestHeaders()), responseType);
+	private void addDefaultAuthorization(HttpHeaders requestHeaders) {
```

Comment:
**[Nit]** Since this is a private 1-liner, may as well inline it to keep the eyes from having to jump down when scanning the code.

Reply:
Adjust it in the merge.